core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -198,6 +224,10 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging {
     getOption(key).getOrElse(defaultValue)
   }
 
+  private[spark] def get[T](entry: ConfigEntry[T]): T = {
```

Comment:
can you add java doc, mention something like:
- for internal use only
- return value as the type specified by config
- if the config is not marked as optional and there is no default value, throw exception.